### DISCUSSION TITLE
Future call serialization error

### DISCUSSION THREAD

**Discussion by eamnicoletti** [serverpod/serverpod#880] [Q&A] [answered]
I'm trying out Future Calls. Everything is going fine, but when I pass a `SerializableEntity object` I get the following issue when it trying to deserialize the database object past the scheduled point:

```
2023-05-06 22:03:46.101814Z Internal server error. Failed to connect to database in future call manager.
FormatException: No deserialization found for type SerializableEntity
#0      SerializationManager.deserialize (package:serverpod_serialization/src/serialization.dart:79:5)
#1      Protocol.deserialize (package:fiscaliza_server/src/generated/protocol.dart:1087:18)
#2      SerializationManager.decode (package:serverpod_serialization/src/serialization.dart:40:12)
#3      FutureCallManager._checkQueue (package:serverpod/src/server/future_call_manager.dart:123:42)
<asynchronous suspension>

Local stacktrace:
#0      FutureCallManager._checkQueue (package:serverpod/src/server/future_call_manager.dart:146:36)
<asynchronous suspension>
```

P.S.: I registered the Future Call.

My scheduling code example (called in the sign in method):
```
session.serverpod.futureCallWithDelay(
  authFutureCall, // const String
  authKey, // Authkey() object
  Duration(minutes: 1),
);
```

I'm actually trying to make an _expired login_ feat, but I also noticed that in the Futurecall `invoke` method, the `authenticatedUserId` is `null` for the session in that Futurecall. And without the authenticatedUserId or a SerializableEntity object, I can't sign the user out properly.

Serverpod version 1.1.0

**Comment by vlidholt** [comment]
I'm not exactly sure what is going on here, but instead of passing the auth key, it may be smarter to pass a custom serializable object that contains the users id. Future calls are detached from user sessions. So when the call is invoked, Serverpod has no knowledge of who was signed in when the call was made.

**Reply by eamnicoletti** [reply]
Passing my custom user object was my first option, but it didn't work as well, so I´d tried with a "native" one like `AuthKey`. But as I said, had the same problem.

P.S.: The object has been correctly inserted into the database future call table.

**Comment by Schamppu** [comment]
Is there any fix to this? I'm also running to the same problem. If I pass a null to the SerializableEntity, it works fine. But immediately when I pass anything (even my custom user class that's serialized by Serverpod) it throws this error.

**Reply by eamnicoletti** [reply, marked as answer]
With the new documentation things have become clearer.

Change this:
```
class ExampleFutureCall extends FutureCall {
  @override
  Future<void> invoke(Session session, SerializableEntity? object) async {
     // Do something interesting in the future here.
  }
```
  
To this:
```
class ExampleFutureCall extends FutureCall<MyModelEntity> {
  @override
  Future<void> invoke(Session session, MyModelEntity? object) async {
    // Do something interesting in the future here.
  }
}
```
